### PR TITLE
docs(samples): Add Cloud Spanner Batch DML sample

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: samples/dml.js [hunk 1]
```diff
@@ -529,6 +529,61 @@ async function deleteUsingPartitionedDml(instanceId, databaseId, projectId) {
...
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
```
Comment (JustinBeckwith) [inline samples/dml.js]: do we need to explicitly define the projectId here?

File: samples/dml.js [hunk 2]
```diff
@@ -529,6 +529,61 @@ async function deleteUsingPartitionedDml(instanceId, databaseId, projectId) {
...
+  };
+
+  const update = {
+    sql: `UPDATE Albums SET MarketingBudget = MarketingBudget * 2
+      WHERE SingerId = 1 and AlbumId = 3`,
+  };
+
+  const dmlStatements = [insert, update];
+
+  database.runTransaction(async (err, transaction) => {
```
Comment (JustinBeckwith) [inline samples/dml.js]: can we use `async/await` here?

File: samples/dml.js [hunk 3]
```diff
@@ -529,6 +529,61 @@ async function deleteUsingPartitionedDml(instanceId, databaseId, projectId) {
...
+    }
+    try {
+      const rowCounts = await transaction.batchUpdate(dmlStatements);
+
+      console.log(
+        `Successfully executed ${rowCounts.length} SQL statements using Batch DML.`
+      );
+      await transaction.commit();
+    } catch (err) {
+      console.error('ERROR:', err);
```
Comment (JustinBeckwith) [inline samples/dml.js]: Can we re-throw here?

File: samples/dml.js (hunk 2 above)
Comment (callmehiphop) [reply samples/dml.js]: Here's an example of that - https://github.com/googleapis/nodejs-spanner/blob/master/system-test/spanner.ts#L3836

File: samples/dml.js [hunk 4]
```diff
@@ -529,6 +529,61 @@ async function deleteUsingPartitionedDml(instanceId, databaseId, projectId) {
...
+
+  const dmlStatements = [insert, update];
+
+  database.runTransaction(async (err, transaction) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    try {
+      const rowCounts = await transaction.batchUpdate(dmlStatements);
```
Comment (callmehiphop) [inline samples/dml.js]: `batchUpdate` returns an array so this should be 

```js
const [rowCounts] = await transaction.batchUpdate(dmlStatements);
```

File: samples/dml.js [hunk 5]
```diff
@@ -529,6 +529,57 @@ async function deleteUsingPartitionedDml(instanceId, databaseId, projectId) {
...
+  };
+
+  const dmlStatements = [insert, update];
+
+  try {
+    await database.runTransactionAsync(async transaction => {
+      const [rowCounts] = await transaction.batchUpdate(dmlStatements);
+      await transaction.commit();
+      console.log(
+        `Successfully executed ${rowCounts.length} SQL statements using Batch DML.`
```
Comment (callmehiphop) [inline samples/dml.js]: Oh whoops, I missed this, but `rowCounts` should be a number

Comment (callmehiphop) [reply samples/dml.js]: Made a mistake, ignore me!

File: samples/dml.js (hunk 1 above)
Comment (jsimonweb) [reply samples/dml.js]: This matches working client instantiation already existing in sample. I'd prefer to address this in a separate PR if possible.

**Pull request merged**